map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -250,9 +265,40 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
         mapView.onLowMemory()
     }
 
+    private fun loadAmenities(amenitiesList: List<MapItem.Annotation>) {
+        amenitiesMarkerList
+                .forEach { marker -> marker.remove() }
+
+        amenitiesList.forEach { mapItem ->
```

Comment:
looks like you're re-adding the markers but not removing them physically from the list.

Reply:
fixed.